packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -64,43 +69,94 @@ async function isLambdaFunctionCodeOnlyChange(
    * which means we don't need the "old" values at all, and we can safely initialize these with just `''`.
    */
   let s3Bucket = '', s3Key = '';
-  let foundCodeDifference = false;
+  let foundDifference = false;
+  let tagUpdates: { [tag: string]: string | TagDeletion } = {};
   // Make sure only the code in the Lambda function changed
   const propertyUpdates = change.propertyUpdates;
   for (const updatedPropName in propertyUpdates) {
     const updatedProp = propertyUpdates[updatedPropName];
-    for (const newPropName in updatedProp.newValue) {
-      switch (newPropName) {
-        case 'S3Bucket':
-          foundCodeDifference = true;
-          s3Bucket = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
-          break;
-        case 'S3Key':
-          foundCodeDifference = true;
-          s3Key = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
-          break;
-        default:
-          return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
-      }
+
+    switch (updatedPropName) {
+      case 'Code':
+        for (const newPropName in updatedProp.newValue) {
+          switch (newPropName) {
+            case 'S3Bucket':
+              foundDifference = true;
+              s3Bucket = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
+              break;
+            case 'S3Key':
+              foundDifference = true;
+              s3Key = await evaluateCfnTemplate.evaluateCfnExpression(updatedProp.newValue[newPropName]);
+              break;
+            default:
+              return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
+          }
+        }
+        break;
+      case 'Tags':
+        /*
+         * Tag updates are a bit odd; they manifest as two lists, are flagged only as
+         * `isDifferent`, and we have to reconcile them.
+         */
+
+        if (updatedProp !== undefined) {
+          foundDifference = foundDifference || updatedProp.isDifferent;
```

Comment:
Hmm. What if we changed the above `if` statement to be `if (updatedProp?.isDifferent) {`? That would allow us to simplify this to just `foundDifference = true;`.

Reply:
I like it!